src/cosalib/qemuvariants.py

```diff
@@ -231,6 +232,10 @@ def mutate_image(self):
         :param callback: callback function for extra processing image
         :type callback: function
         """
+
+        repo = os.path.join(self._workdir, 'tmp/repo')
+        import_ostree_commit(repo, self.build_dir, self.meta)
```

Comment:
Some mutators (called via the callback support in this function) need values from `image.json`, like VMware. A comment SGTM.

Reply:
ahh. I see. https://github.com/coreos/coreos-assembler/blob/3c7694e3fcfe8f6232b1f86f6aa084b5867b3b4d/src/cosalib/ova.py#L91

Still.. Maybe https://github.com/coreos/coreos-assembler/pull/2839/files#r880776011 is the better approach here?

Reply:
or we could just put the `import_ostree_commit()` down in `ova.py` where it is used.

Reply:
Today, yes.  But the way I think about this, access to the `image.json` is going to be commonly used by disk image builds.  Confusingly, there are actually *three* almost completely independent code paths for disk image builds:

qemu, metal: shell script
live: custom "script Python"
vsphere,aws: heavy OO Python subclassing `QemuVariantImage` etc.

So you're right, today of the latter it's just `ova.py` that needs `image.json`. 

But I think *all* disk image builds should have consistent access to it, so it makes sense to do in a common place in the 3rd path, in the same way we do for both the shell script and live paths.

Reply:
I think what's confusing is that this code sits by itself with no comment and you have to just know that some callback function *might* want to extract the `image.json` from the ostree so we need to import the ostree here. It would be way more obvious if we just did the import lazily inside of `extract_image_json()` like I suggested above, OR hoist the ostree import into ova.py (the only place it is needed today).

Reply:
> I think what's confusing is that this code sits by itself with no comment 

OK, the code now has a comment.

> It would be way more obvious if we just did the import lazily inside of extract_image_json() like I suggested above,

I flipped things around, `import_ostree_commit` now always *also* generates `tmp/image.json` since disk image builds in the general case may need both.

Reply:
> I think what's confusing 

You're right this is confusing but I will restate I think the *real* problem is I'm trying to make a slightly nontrivial change in a horribly messy codebase.  To restate, I agree with you, but I think the real fix is to move closer to a [single programming language](https://github.com/coreos/coreos-assembler/issues/2821) for the build system, in which case this problem wouldn't have appeared at all because we wouldn't have 3 different code paths and styles for disk images.